modules/cli/src/main/scala/scala/cli/commands/Repl.scala

```diff
@@ -31,8 +31,34 @@ object Repl extends ScalaCommand[ReplOptions] {
   override def buildOptions(ops: ReplOptions): Some[BuildOptions] = Some {
     import ops.*
     import ops.sharedRepl.*
-    val baseOptions = shared.buildOptions().orExit(ops.shared.logger)
+
+    val ammoniteVersionOpt = ammoniteVersion.map(_.trim).filter(_.nonEmpty)
+    val baseOptions        = shared.buildOptions().orExit(ops.shared.logger)
     baseOptions.copy(
+      scalaOptions = baseOptions.scalaOptions.copy(
+        scalaVersion = baseOptions.scalaOptions.scalaVersion
+          .orElse {
+            val maxScalaVer     = scala.cli.internal.Constants.maxAmmoniteScala3Version
+            val defaultScalaVer = scala.build.internal.Constants.defaultScalaVersion
+            val shouldDowngrade = {
+              def needsDowngradeForAmmonite = {
+                import coursier.core.Version
+                Version(maxScalaVer) < Version(defaultScalaVer)
+              }
+              ammonite.contains(true) &&
+              ammoniteVersionOpt.isEmpty &&
+              needsDowngradeForAmmonite
+            }
+            if (shouldDowngrade) {
+              System.err.println(
```

Comment:
NIT might want to use `shared.logger` here instead of just `System.err`

Reply:
Should be fixed.